lib/Model/Type.cpp

```diff
@@ -508,6 +550,19 @@ verifyImpl(VerifyHelper &VH, const PrimitiveType *T) {
                          and isValidPrimitiveSize(T->PrimitiveKind, T->Size));
 }
 
+model::Identifier makeValidIdentifier(model::Identifier &&I) {
+  for (auto &C : I) {
+    if (C != '_' and not isalnum(C))
+      C = '_';
+  }
+  return std::move(I);
```

Comment:
This can generate invalid identifiers starting with `_` or `CReservedKeywords`

Reply:
It can, and so does the factory you already have included in the code. I'll try to come up with a fix for both of them.

Reply:
Done, take a look.